### PR TITLE
Lambda Layers

### PR DESCRIPTION
**Pull request by amarjandu**
Reduce Lambda size by having common python dependency layer.
Removes Requirements.txt for API/Daemons, decreases size due to not downloading/packaging dependencies.

Deployment script added to check status of requirements.txt in AWS-Terraform-Bucket by comparing it to the local requirements.txt; upon finding changes between the files, it creates/uploads the dependencies.zip and creates a layer with the appropriate folder structure to be used as a layer.

Deployment process includes this script on deploying an entire DSS or just an isolated component, through the make commands, the layer-version ARN is fetched from the aws-lambda-api, and added to the Chalice config.json during the build script execution. 

Resolves: #2118

### CODE REVIEW

**Comment by amarjandu** [comment]
@Bento007 you might have to remove the webooks for the removed requirements.txt files if approved. see below.

File: scripts/generate_upload_requirements_layer.sh [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+if [[ $aws_req_hash != $local_req_hash ]]; then
+ echo "checksum missmatch, uploading new ${aws_zip_key}"
+ upload
+else
+ echo "requirements have not changed, no need to update"
+ exit 0
+fi
+}
+
+function upload() {
```
Comment (xbrianh) [inline scripts/generate_upload_requirements_layer.sh]: Please update the formatting in this file to be consistent with existing DSS bash scripting: 4 space indentation for functions and flow control.

File: scripts/generate_upload_requirements_layer.sh [hunk 2]
```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env bash
+# TODO make this description not suck
+# This script is used to generate a requirements layer for chalice to use
+
+set -euo pipefail
+
+if [[ -z $DSS_DEPLOYMENT_STAGE ]]; then
+    echo 'Please run "source environment" in the data-store repo root directory before running this command'
```
[the remark this answers is not in the document]
Comment (amarjandu) [reply scripts/generate_upload_requirements_layer.sh]: removed.

File: scripts/generate_upload_requirements_layer.sh (hunk 1 above)
Comment (amarjandu) [reply scripts/generate_upload_requirements_layer.sh]: resolved

**Pull request merged**